src/algebra/category/CommRing/adjunctions.lean

```diff
@@ -41,12 +41,22 @@ def free : Type u ⥤ CommRing.{u} :=
 @[simp] lemma free_map_coe {α β : Type u} {f : α → β} :
   ⇑(free.map f) = rename f := rfl
 
+namespace adj
+-- The next two definitions are (unfortunate) "implementation details", helping the elaborator / typeclass search.
+
+/-- Version of `mv_polynomial.hom_equiv` for bundled commutative rings. -/
+def hom_equiv (X : Type u) (R : CommRing) : (mv_polynomial X ℤ →+* R) ≃ (X → R) := hom_equiv R X
```

Comment:
This clearly indicates that something is wrong. Do you understand the error in `adj` that occurs without this definition?